endgame_models/get_differences.py

```diff
@@ -0,0 +1,96 @@
+from dataclasses import dataclass
+from typing import Optional, TypeVar
+
+from typing_extensions import Type
+
+from .models import BaseInitialParams
+
+Flat = str | float | int | bool
+
+JSONType = Flat | list["JSONType"] | dict[str, "JSONType"] | tuple["JSONType", ...]
+DictType = dict[str, JSONType]
+ListType = list[JSONType]
+TupleType = tuple[JSONType, ...]
+
+
+@dataclass
+class ReadOnlyDiff:
+    original: JSONType
+    new: JSONType
+
+
+OutputReadOnly = dict[str, "ReadOnlyDiff | OutputReadOnly"]
```

Comment:
Ha, that's a funny one. You probably wanted to write `dict[str, ReadOnlyDiff | "OutputReadOnly"]` or not? If not - how  why?